src/app/clusters/access-control-server/ArlEncoder.cpp

```diff
@@ -0,0 +1,152 @@
+{
+    size_t count = source.size();
+    if (count > 0 && count <= destinationCount)
+    {
+        for (size_t i = 0; i < count; i++)
+        {
+            auto restriction = source[i];
+            ReturnErrorOnFailure(ArlEncoder::Convert(restriction.restrictionType, destination[i].type));
+
+            if (restriction.id.HasValue())
+            {
+                destination[i].id.SetNonNull(restriction.id.Value());
+            }
+        }
+    }
+    else
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+} // namespace
+
+namespace chip {
+namespace app {
+
+CHIP_ERROR ArlEncoder::Convert(Clusters::AccessControl::AccessRestrictionTypeEnum from,
+                               Access::AccessRestrictionProvider::Type & to)
+{
+    switch (from)
+    {
+    case StagingRestrictionType::kAttributeAccessForbidden:
+        to = AccessRestrictionProvider::Type::kAttributeAccessForbidden;
+        break;
+    case StagingRestrictionType::kAttributeWriteForbidden:
+        to = AccessRestrictionProvider::Type::kAttributeWriteForbidden;
+        break;
+    case StagingRestrictionType::kCommandForbidden:
+        to = AccessRestrictionProvider::Type::kCommandForbidden;
+        break;
+    case StagingRestrictionType::kEventForbidden:
+        to = AccessRestrictionProvider::Type::kEventForbidden;
+        break;
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ArlEncoder::Convert(Access::AccessRestrictionProvider::Type from,
+                               Clusters::AccessControl::AccessRestrictionTypeEnum & to)
+{
+    switch (from)
+    {
+    case AccessRestrictionProvider::Type::kAttributeAccessForbidden:
+        to = StagingRestrictionType::kAttributeAccessForbidden;
+        break;
+    case AccessRestrictionProvider::Type::kAttributeWriteForbidden:
+        to = StagingRestrictionType::kAttributeWriteForbidden;
+        break;
+    case AccessRestrictionProvider::Type::kCommandForbidden:
+        to = StagingRestrictionType::kCommandForbidden;
+        break;
+    case AccessRestrictionProvider::Type::kEventForbidden:
+        to = StagingRestrictionType::kEventForbidden;
+        break;
+    default:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ArlEncoder::CommissioningEncodableEntry::Encode(TLV::TLVWriter & writer, TLV::Tag tag) const
+{
+    ReturnErrorOnFailure(Stage());
+    ReturnErrorOnFailure(mStagingEntry.Encode(writer, tag));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ArlEncoder::EncodableEntry::EncodeForRead(TLV::TLVWriter & writer, TLV::Tag tag, FabricIndex fabric) const
+{
+    ReturnErrorOnFailure(Stage());
+    ReturnErrorOnFailure(mStagingEntry.EncodeForRead(writer, tag, fabric));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ArlEncoder::EncodableEntry::EncodeForWrite(TLV::TLVWriter & writer, TLV::Tag tag) const
```

Comment:
I don't think this EncodeForWrite method is used anywhere.   Would suggest removing it for now.

Reply:
+1

Reply:
done